R/bs-theme-preview.R

```diff
@@ -459,28 +435,35 @@ bs_theme_get_variables <- function(varnames) {
   stats::setNames(values, varnames)
 }
 
-take_and_use_values <- function(changed_vals, func, use_all_vals = FALSE) {
-  if (use_all_vals) {
-    call_vals <- changed_vals
-    return_vals <- NULL
-  } else {
-    args_src <- names(changed_vals)
-    args_dest <- names(formals(func))
-    call_vals <- changed_vals[args_src %in% args_dest]
-    return_vals <- changed_vals[!args_src %in% args_dest]
-  }
 
-  if (length(call_vals)) {
-    # Call the function (modifying global state)
-    do.call(func, call_vals)
-    # Print the code for the user
-    func_name <- substitute(func)
-    print(rlang::expr(`!!`(func_name)(!!!call_vals)))
-  }
 
-  return_vals
+echo_and_eval_theme_code <- function(theme, vals) {
+  funcs <- list(
+    bs_base_colors = bs_base_colors,
+    bs_accent_colors = bs_accent_colors,
+    bs_fonts = bs_fonts
+  )
```

Comment:
This could probably be simplified to use the new `bs_theme_update()`. If we can, we should consider not exporting `bs_base_colors`, `bs_accent_colors`, and `bs_fonts`